main/webapp/modules/core/scripts/reconciliation/recon-dialog-2.js

```diff
@@ -0,0 +1,142 @@
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+ */
+
+function ReconDialog2(column, selectedServiceRecordindex, serviceRecords, record, previousService) {
+  this._column = column;
+  this._serviceRecords = serviceRecords;
+  this._selectedServiceRecordIndex = selectedServiceRecordindex;
+  this._previousService = previousService;
+  this._record = record;
+
+  this._createDialog();
+}
+
+ReconDialog2.prototype._createDialog = function() {
+  var self = this;
+  var dialog = $(DOM.loadHTML("core", "scripts/reconciliation/recon-dialog-2.html"));
+
+  this._elmts = DOM.bind(dialog);
+  this._elmts.dialogHeader.text($.i18n('core-recon/recon-col',this._column.name ));
+  this._elmts.reconcileButton.html($.i18n('core-buttons/start-recon'));
+  this._elmts.cancelButton.html($.i18n('core-buttons/cancel'));
+  this._elmts.backButton.html($.i18n('core-buttons/previous'));
+
+  this._elmts.reconcileButton.on('click',function() { self._onOK(); });
+  this._elmts.cancelButton.on('click',function() { self._dismiss(); });
+  this._elmts.backButton.on('click',function() { self._previousDialog(self._column, self._previousService); });
+
+
+  this._level = DialogSystem.showDialog(dialog);
+  this._selectService(self._record);
+};
+
+
+ReconDialog2.prototype._previousDialog = function(column, previousService) {
+  this._dismiss();
+  new ReconDialog(column, previousService);
+}
+
+
+ReconDialog2.prototype._onOK = function() {
+
+  if (this._selectedServiceRecordIndex >= 0) {
+    var record = this._serviceRecords[this._selectedServiceRecordIndex];
+    if (record.handler) {
+      record.handler.start();
+    }
+  }
+  this._dismiss();
+};
+
+
+ReconDialog2.prototype._dismiss = function() {
+  for (var i = 0; i < this._serviceRecords.length; i++) {
+    var record = this._serviceRecords[i];
+    if (record.handler) {
+      record.handler.dispose();
+    }
+  }
+  this._serviceRecords = null;
+
+  DialogSystem.dismissUntil(this._level - 1);
+};
+
+
+ReconDialog2.prototype._cleanDialog = function() {
+  for (var i = 0; i < this._serviceRecords.length; i++) {
+    var record = this._serviceRecords[i];
+    if (record.handler) {
+      record.handler.deactivate();
+    }
+    record.selector.remove();
+  }
+  this._serviceRecords = [];
+  this._selectedServiceRecordIndex = -1;
+};
+
+
+ReconDialog2.prototype._selectService = function(record) {
+
+  for (var i = 0; i < this._serviceRecords.length; i++) {
+    if (record === this._serviceRecords[i]) {
+      if (i !== this._selectedServiceRecordIndex) {
+        if (this._selectedServiceRecordIndex >= 0) {
+          var oldRecord = this._serviceRecords[this._selectedServiceRecordIndex];
+          if (oldRecord.handler) {
+            oldRecord.selector.removeClass("selected");
+            oldRecord.handler.deactivate();
+          }
+        }
+
+        record.selector.addClass("selected");
```

Comment:
This code block feels overly complicated to me. It seems that you are manually re-implementing the logic of selecting a service and un-selecting any other service whenever the user clicks on a record. Could it be that this is a left over from the code from the previous dialog, which did not have radio buttons? If the radio buttons are correctly grouped together (by having the same `name` attribute) then the browser should already enforce the fact that selecting a service unselects any previously selected service). I think it would be preferable to rely on that mechanism.